crates/vault-registry/src/types.rs

```diff
@@ -441,40 +441,59 @@ impl<T: Config> RichVault<T> {
         // Note: slashing of collateral must be called where this function is called (e.g. in Redeem)
     }
 
+    pub(crate) fn slash_collateral_and_transfer(&mut self, amount: Collateral<T>) -> DispatchResult {
+        self.slash_collateral(amount)?;
+        let vault_id = self.id();
+        let liquidation_vault_id = Pallet::<T>::liquidation_vault_account_id();
+        ext::collateral::release::<T>(&vault_id, amount)?;
+        ext::collateral::transfer::<T>(&vault_id, &liquidation_vault_id, amount)?;
+        ext::collateral::lock::<T>(&liquidation_vault_id, amount)?;
+        Ok(())
+    }
+
     pub(crate) fn liquidate<V: UpdatableVault<T>>(
         &mut self,
         liquidation_vault: &mut V,
-        status: VaultStatus,
     ) -> Result<Collateral<T>, DispatchError> {
-        let backing_collateral = self.data.backing_collateral;
-
-        // we liquidate at most SECURE_THRESHOLD * collateral.
+        // we liquidate at most SECURE_THRESHOLD * collateral
+        // this value is the amount of collateral held for the issued + to_be_issued
         let liquidated_collateral = self.get_used_collateral()?;
 
         // amount of tokens being backed
         let collateral_tokens = self.backed_tokens()?;
 
-        let to_slash = Pallet::<T>::calculate_collateral(
+        // (liquidated_collateral * (collateral_tokens - to_be_redeemed_tokens)) / collateral_tokens
+        let collateral_for_issued_and_to_be_issued = Pallet::<T>::calculate_collateral(
```

Comment:
this name is incorrect - `liquidated_collateral` is the collateral for issued and to be issued

Reply:
Are you sure? In my calculations this was to ascertain the amount required for the backed tokens excluding the `to_be_redeemed`.

Reply:
`liquidated_collateral` is the complete collateral that backs `issued` & `to_be_issued`. The variable that you call `collateral_for_issued_and_to_be_issued` should be called `collateral_for_issued_and_to_be_issued_excluding_to_be_redeemed` if you want to spell it out. That's the amount that needs to be slashed right away in this function - hence the old name `to_slash`

Reply:
Ah okay yeah this is mostly semantics then, I'll rename to something more applicable.